lib/backends/system-manager-backend.js

```diff
@@ -49,13 +51,34 @@ class SystemManagerBackend extends KVBackend {
       client = this._clientFactory(factoryArgs)
     }
     try {
-      const data = await client
-        .getParameter({
-          Name: key,
-          WithDecryption: true
-        })
+      const params = {
+        Name: key,
+        WithDecryption: false
+      }
+      const check = await client
+        .getParameterHistory(params)
         .promise()
-      return data.Parameter.Value
+
+      while (check.NextToken) {
+        params.NextToken = check.NextToken
+        await client.getParameterHistory(params).promise()
+      }
+      const timetocompare = new Date(Date.now() - this._intervalPolling)
+
+      if (check.Parameters[check.Parameters.length - 1].LastModifiedDate > timetocompare || !this._checkUpdated) {
```

Comment:
Hi @moolen. I think your idea is better, thanks for the suggestion.

I did this draw and please tell me if i got your idea:
![image](https://user-images.githubusercontent.com/11686779/112057972-be2d4780-8b38-11eb-9c0a-d32c77408c2f.png)

1- The ParameterMetadaService is an infinite loop that store (in memory?) the parameter history
2- The SystemManagerBackend consume the API explained in 1.
3. If the parameter has changed, SystemManagerBackend fetch the new value

Is there any similar code in this project that i can be based on?

Reply:
Hey! yeah, that's pretty much what i've thought of! Yes, it probably needs to be in-memory (hence the load test). 
> Is there any similar code in this project that i can be based on?

There's nothing i'm aware of. I imagine a

```javascript
cass PMS {
  constructor(interval){
    this._interval = interval;
    this.data = {};
  }

  // this does the hard work
  fetchHistory(){
    /*
    for every N seconds; do:
      for each account/region combination that was previously seen; do:
        1. fetch history of all parameters
        2. store them in this.data 
    */
  }
  
  // this is the cheap lookup
  hasChanged(account, region, param, since){
    // the lookup key is important, the controller can handle multiple accounts/roles
    if data[`${account}/${role}`][param].LastModifiedDate > since {
      return true
    }
    return false
  }
}
```